src/backend/distributed/connection/connection_management.c

```diff
@@ -766,3 +772,53 @@ AfterXactHostConnectionHandling(ConnectionHashEntry *entry, bool isCommit)
 		}
 	}
 }
+
+
+/*
+ * SetCitusNoticeProcessor sets the NoticeProcessor to DefaultCitusNoticeProcessor
+ */
+void
+SetCitusNoticeProcessor(MultiConnection *connection)
+{
+	PQsetNoticeProcessor(connection->pgConn, DefaultCitusNoticeProcessor,
+						 connection);
+}
+
+
+/*
+ * SetCitusNoticeLevel is used to set the notice level for distributed
+ * queries.
+ */
+void
+SetCitusNoticeLevel(int level)
+{
+	CitusNoticeLogLevel = level;
+}
+
+
+/*
+ * UnsetCitusNoticeLevel sets the CitusNoticeLogLevel back to
+ * its default value.
+ */
+void
+UnsetCitusNoticeLevel()
+{
+	CitusNoticeLogLevel = DEFAULT_CITUS_NOTICE_LEVEL;
+}
+
+
+/*
+ * DefaultCitusNoticeProcessor is used to redirect worker notices
+ * from logfile to console.
+ */
+static void
+DefaultCitusNoticeProcessor(void *arg, const char *message)
+{
+	MultiConnection *connection = (MultiConnection *) arg;
+	char *nodeName = connection->hostname;
+	uint32 nodePort = connection->port;
+
+	ereport(CitusNoticeLogLevel, (errmsg("%s", message),
```

Comment:
it seems we put an extra newline after messages. Could we remove it like we do in `ReportResultError`? 

Would also be nice to remove the "INFO: " prefix from the message, since otherwise we'll give that twice.